app/src/main/java/bg/zahov/app/ui/exercise/info/ExerciseInfoScreen.kt

```diff
@@ -0,0 +1,330 @@
+                            clipInOverlayDuringTransition = OverlayClip(RoundedCornerShape(8.dp))
+                        )
+                        .background(MaterialTheme.colorScheme.secondary, RoundedCornerShape(8.dp))
+                        .clip(RoundedCornerShape(8.dp))
+                ) {
+
+                    ExerciseChartInfo(
+                        data = chart,
+                        modifier = Modifier.sharedElement(
+                            state = rememberSharedContentState(key = chart.text),
+                            animatedVisibilityScope = this@AnimatedContent,
+                        ),
+                        onClick = onConfirmClick
+
+                    )
+                    CommonLineChart(modifier = Modifier.width(300.dp), data = chart)
+                }
+            }
+        }
+    }
+}
+
+@Preview
+@Composable
+fun ExerciseChartInfo(
+    modifier: Modifier = Modifier,
+    data: LineChartData = LineChartData(),
+    onClick: () -> Unit = {}
+) {
+    Column(
+        modifier = modifier
+            .padding(20.dp)
+            .clickable { onClick() }
+            .background(MaterialTheme.colorScheme.secondary, RoundedCornerShape(8.dp)),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Text(
+            modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp),
+            text = data.text, color = MaterialTheme.colorScheme.onSecondary,
+            style = MaterialTheme.typography.titleMedium
+        )
+        Text(
+            modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp),
+            text = "${stringResource(R.string.personal_records)}: ${data.maxValue}",
+            color = MaterialTheme.colorScheme.onSecondary,
+            style = MaterialTheme.typography.titleMedium
+        )
+    }
+}
+
+@Preview
+@Composable
+fun ExerciseHistoryCard(
+    modifier: Modifier = Modifier,
+    data: ExerciseHistoryInfo = ExerciseHistoryInfo()
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(16.dp)
+            .background(MaterialTheme.colorScheme.secondary, RoundedCornerShape(16.dp)),
+        verticalArrangement = Arrangement.spacedBy(20.dp),
+    ) {
+        Text(
+            modifier = Modifier.padding(start = 16.dp),
+            text = data.workoutName,
+            color = MaterialTheme.colorScheme.onSecondary,
+            fontWeight = FontWeight.Bold,
+            style = MaterialTheme.typography.titleMedium
+        )
+        Text(
+            modifier = Modifier.padding(start = 16.dp),
+            text = data.lastPerformed,
+            color = MaterialTheme.colorScheme.onSecondary,
+            fontWeight = FontWeight.Bold,
+            style = MaterialTheme.typography.titleMedium
+        )
+
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceAround
+        ) {
+            Column {
+                Text(
```

Comment:
Those texts are in `Column { Row { Column { Text() } } }` structure. Perhaps it's a good idea to use maxLines and ellipsize for them?

Reply:
Its is column { Row { Column {text, text} Column {text, text}}. I wanted to have two columns with two textfield which are on both ends of the row. I am not really sure how to do it another way

Reply:
I see no issue in the arrangement. I just pointed out that it indicates multiple texts on the same row. 
The question is if you think those texts should be elipsized and constraint in any way. Did you test this screen with large text system setting?